modules/packages/Zarr.chpl

```diff
@@ -367,50 +367,21 @@ module Zarr {
 
     :arg dimCount: Dimensionality of the zarr array
 
+    :arg bloscThreads: The number of threads to use during compression (default=1)
+
+    :arg targetLocales: The locales to use for reading the array in the shape the
+      array will be distributed
   */
   proc readZarrArray(directoryPath: string, type dtype, param dimCount: int, bloscThreads: int(32) = 1, targetLocales: [] locale = Locales) throws {
     var md = getMetadata(directoryPath);
     validateMetadata(md, dtype, dimCount);
-    // Size and shape tuples
-    var totalShape, chunkShape : dimCount*int;
-    var chunkCounts: dimCount*int;
-    var totalRanges,chunkRanges: dimCount*range(int);
+    var totalRanges: dimCount*range(int);
     for i in 0..<dimCount {
-      totalShape[i] = md.shape[i];
-      chunkShape[i] = md.chunks[i];
-      chunkCounts[i] = ceil(totalShape[i]:real / chunkShape[i]:real) : int;
-      totalRanges[i] = 0..<totalShape[i];
-      chunkRanges[i] = 0..<chunkCounts[i];
+      totalRanges[i] = 0..<md.shape[i];
```

Comment:
Is it possible for the ``shape`` to be strided or aligned? If so, you'd need to change the ``range(int)`` type above.

Reply:
As well as the domain type below.

Reply:
it is not for zarr 2.0